controller_manager:
  ros__parameters:
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    io_and_status_controller:
      type: ur_controllers/GPIOController

    speed_scaling_state_broadcaster:
      type: ur_controllers/SpeedScalingStateBroadcaster

    force_torque_sensor_broadcaster:
      type: force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    scaled_joint_trajectory_controller:
      type: ur_controllers/ScaledJointTrajectoryController

    forward_velocity_controller:
      type: velocity_controllers/JointGroupVelocityController

    forward_position_controller:
      type: position_controllers/JointGroupPositionController

    force_mode_controller:
      type: ur_controllers/ForceModeController

    freedrive_mode_controller:
      type: ur_controllers/FreedriveModeController

    passthrough_trajectory_controller:
      type: ur_controllers/PassthroughTrajectoryController

    ur_configuration_controller:
      type: ur_controllers/URConfigurationController

speed_scaling_state_broadcaster:
  ros__parameters:
    state_publish_rate: 100.0
    tf_prefix: "arm_"

io_and_status_controller:
  ros__parameters:
    tf_prefix: "arm_"

force_torque_sensor_broadcaster:
  ros__parameters:
    sensor_name: arm_tcp_fts_sensor
    state_interface_names:
      - force.x
      - force.y
      - force.z
      - torque.x
      - torque.y
      - torque.z
    frame_id: arm_tool0
    topic_name: ft_data

joint_trajectory_controller:
  ros__parameters:
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_joint
      - arm_wrist_1_joint
      - arm_wrist_2_joint
      - arm_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: true
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      arm_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      arm_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      arm_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }

scaled_joint_trajectory_controller:
  ros__parameters:
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_joint
      - arm_wrist_1_joint
      - arm_wrist_2_joint
      - arm_wrist_3_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false
    allow_nonzero_velocity_at_trajectory_end: true
    constraints:
      stopped_velocity_tolerance: 0.2
      goal_time: 0.0
      arm_shoulder_pan_joint: { trajectory: 0.2, goal: 0.1 }
      arm_shoulder_lift_joint: { trajectory: 0.2, goal: 0.1 }
      arm_elbow_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_1_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_2_joint: { trajectory: 0.2, goal: 0.1 }
      arm_wrist_3_joint: { trajectory: 0.2, goal: 0.1 }
    speed_scaling_interface_name: arm_speed_scaling/speed_scaling_factor

forward_velocity_controller:
  ros__parameters:
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_joint
      - arm_wrist_1_joint
      - arm_wrist_2_joint
      - arm_wrist_3_joint
    interface_name: velocity

forward_position_controller:
  ros__parameters:
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_joint
      - arm_wrist_1_joint
      - arm_wrist_2_joint
      - arm_wrist_3_joint

force_mode_controller:
  ros__parameters:
    tf_prefix: "arm_"

freedrive_mode_controller:
  ros__parameters:
    tf_prefix: "arm_"

passthrough_trajectory_controller:
  ros__parameters:
    tf_prefix: "arm_"
    joints:
      - arm_shoulder_pan_joint
      - arm_shoulder_lift_joint
      - arm_elbow_joint
      - arm_wrist_1_joint
      - arm_wrist_2_joint
      - arm_wrist_3_joint
    state_interfaces:
      - position
      - velocity
    speed_scaling_interface_name: arm_speed_scaling/speed_scaling_factor

ur_configuration_controller:
  ros__parameters:
    tf_prefix: "arm_"

